site/app/views/grading/popup_refactor/RubricGraderView.php

```diff
@@ -0,0 +1,197 @@
+     * @var bool
+     * True if the current gradeable has peer grading.
+     */
+    private $is_peer_gradeable;
+
+    /**
+     * @var bool
+     * True if the current gradeable has teams.
+     */
+    private $is_team_gradeable;
+
+    /**
+     * @var string
+     *
+     * The access mode for the current user for this gradeable.
+     * Possible Values:
+     *  - "unblind" - Nothing about students is hidden.
+     *  - "single"  - For peer grading or for full access grading's Anonymous Mode. Graders cannot see
+     *               who they are currently grading.
+     *  - "double"  - For peer grading. In addition to blinded peer graders, students cannot
+     *               see which peer they are currently grading.
+     */
+    private $blind_access_mode = "";
+
+    // ---------------------------------
+
+
+    // ---
+
+
+    // ---------------------------------
+
+    /**
+     * Creates the Rubric Grading page visually.
+     * This function is called in reateMainRubricGraderPage of RubricGraderController.php.
+     *
+     * @param Gradeable $gradeable - The current gradeable.
+     * @param GradedGradeable $current_submission - The current submission we are looking at.
+     * @param Gradeable $gradeable - The current Gradeable.
+     * @param string $sort_type - The current way we are sorting students. Used to create the header.
+     * @param string $sort_direction -  Either "ASC" or "DESC" for ascending or descending sorting order.
+     *     Used to create the header.
+     * @param bool $is_peer_gradeable - True if the gradeable has peer grading.
+     * @param bool $is_team_gradeable - True if the gradeable is a team gradeable.
+     * @param string $blind_access_mode - Either "unblind", "single", or "double". See above for details.
+     * @param string $details_url - URL of the details page for this Gradeable.
+     *
+     */
+    public function createRubricGradeableView(
+        Gradeable $gradeable,
+        GradedGradeable $current_submission,
+        string $sort_type,
+        string $sort_direction,
+        bool $is_peer_gradeable,
+        bool $is_team_gradeable,
+        string $blind_access_mode,
+        string $details_url
+    ) {
+        $this->setMemberVariables($gradeable, $current_submission, $is_peer_gradeable, $is_team_gradeable, $blind_access_mode);
+
+        $this->createBreadcrumbHeader($sort_type, $sort_direction);
+
+        $this->addCSSs();
+        $this->addJavaScriptCode();
+
+        $page_html = <<<HTML
+                <div class="content" id="electronic-gradeable-container">
+                    <div class="content-items-container">
+                    <div class="content-item content-item-right">
+                HTML;
+
+        $page_html .= $this->renderNavigationBar($details_url);
+
+        return $page_html . <<<HTML
+                                         </div>
+                                     </div>
+                                </div>
+            HTML;
```

Comment:
The indentation is pretty weird here.  In reality, the indentation of the final HTML sent to the browser doesn't matter, so it would be fine to just return the string `</div></div></div>`.  I do question why we are manipulating HTML in PHP in the first place though.  This kind of stuff is exactly why we have Twig.

Reply:
I got this string from the original ElectronicGraderView.php. I just fixed the indenting. I could totally move this string to its own Twig file, but if I did that, I would have to call a Twig file within a twig file. Do you have a good link of an example of the syntax of this?

Reply:
Discussed in person before the meeting.